src/main/java/pl/project13/maven/git/GitCommitIdMojo.java

```diff
@@ -52,24 +54,19 @@
 import java.util.TimeZone;
 
 /**
- * Goal which puts git build-time information into property files or maven's properties.
+ * Puts git build-time information into property files or maven's properties.
  *
- * @goal revision
- * @phase initialize
- * @requiresProject
- * @threadSafe true
  * @since 1.0
  */
-@SuppressWarnings({"JavaDoc"})
+@Mojo(name = "revision", defaultPhase = LifecyclePhase.INITIALIZE, threadSafe = true)
```

Comment:
great to move away from these comment based config. I failed once trying to move away in the past, can't remember why. :+1: